File: Day357_2025/Q357_GFG.java
import java.util.*;

class Q357_GFG {
    public ArrayList<Integer> cntInRange(int[] arr, int[][] queries) {
        ArrayList<Integer> ans = new ArrayList<>();

        Arrays.sort(arr);

        for (int[] q : queries) {
            int a = q[0];
            int b = q[1];

            int left = lowerBound(arr, a);
            int right = upperBound(arr, b);

            ans.add(right - left);
        }

        return ans;
    }

    private int lowerBound(int[] arr, int target) {
        int l = 0, r = arr.length;
        while (l < r) {
            int mid = l + (r - l) / 2;
            if (arr[mid] < target)
                l = mid + 1;
            else
                r = mid;
        }
        return l;
    }

    private int upperBound(int[] arr, int target) {
        int l = 0, r = arr.length;
        while (l < r) {
            int mid = l + (r - l) / 2;
            if (arr[mid] <= target)
                l = mid + 1;
            else
                r = mid;
        }
        return l;
    }
}

File: Day357_2025/Q357_GFG.cpp
class Solution {
  public:
    vector<int> cntInRange(vector<int> &arr, vector<vector<int>> &queries) {
        // Q. Elements in range [a, b]
        // code here
        int n=queries.size();
        vector<int> ans(n, 0);
      
        sort(arr.begin(), arr.end());
      
        int i=0;
        for(vector<int> &v: queries){
            int l = lower_bound(arr.begin(), arr.end(), v[0]) - arr.begin();
            int r = upper_bound(arr.begin(), arr.end(), v[1]) - arr.begin();
            ans[i++] = r-l;
        }
      
        return ans;
    }
};